org.eclipse.jdt.ui.tests.refactoring/resources/ExtractTemp/canExtract/A_test119_out.java

```diff
@@ -0,0 +1,11 @@
+package p; //5, 28, 8, 38
+
+class A {
+	public int m(int[] arr) {
+		int length= arr.length;
```

Comment:
Suppose we add a local declaration 'length=s.length()' before the if statement, it will result in nullpointer exception.  However, if we add the declaration after the else statement (just before the declaration of z, we cannot replace the occurence of 's.length()' within the else statement with the new variable. 

Although it is possible to declare the variable (but without initialization) before the if statement, it could be clumsy.  There is no guarantee whether the variable has been initialized when it is assessed: Initializing it on the else branch only may not guarantee that it has been initialized after the whole if statement.  To this end, we should initialized the same variable multiple times (first on the else branch, and then after the whole if statement).  I don't think it is better than employing different variables to represent the occurrences within else statements and those outside the statement, respectively.

Reply:
A  possible solution here is: replace the expressions within the else statement only （as what the commit does）.

A better solution: Don't do anything here. It is odd to delcare (and initialize) a variable and the only use of the variable is to intialize another variable in the form of VariableA=VarialbeB.

`		int length=s.length();
                int k = length;
		System.out.println(k);`